src/index.js

```diff
@@ -164,10 +167,15 @@ document.addEventListener("DOMContentLoaded", () => {
     const decodedSortFields = JSON.parse(decode(sortFields));
     // Update solrClient and request a new solr query
     const paramesDict = { 'searchFields': decodedSearchFields, 'sortFields': decodedSortFields };
-
     // Use solrClient built-in functions
     // set initial query. This function would not send a query.
+    // This APIs could only merge field values rather than collapse
     solrClient.setInitialQuery(paramesDict);
+    // set initial collapse
+    decodedSearchFields
+      .forEach((field) => {
+        solrClient.setCollapse(field.field, field.collapse);
```

Comment:
Is this right?  Does the `solrClient` already have a `setCollapse` function on it?

Reply:
Ah, ok, I see the existing method in `solrClient`:

```
  setCollapse(field, value) {
    const {query} = this.state;
    const {searchFields} = query;
    const newFields = searchFields
      .map((searchField) => searchField.field === field ? {...searchField, collapse: value} : searchField);
    const payload = {type: "SET_SEARCH_FIELDS", newFields: newFields};
    this.state.query = queryReducer(this.state.query, payload);
    this.onChange(this.state, this.getHandlers());
  }
```

Reply:
So this is going to `setCollapse` on the fields after you copy and paste a link?